httpclient5/src/main/java/org/apache/hc/client5/http/impl/nio/DefaultAsyncClientConnectionOperator.java

```diff
@@ -141,15 +155,25 @@ public void upgrade(
             final ManagedAsyncClientConnection connection,
             final HttpHost host,
             final Object attachment) {
-        upgrade(connection, host, attachment, null);
+        upgrade(connection, host, attachment, null, null);
     }
 
     @Override
     public void upgrade(
             final ManagedAsyncClientConnection connection,
             final HttpHost host,
             final Object attachment,
-            final FutureCallback<ManagedAsyncClientConnection> callback) {
+            final HttpContext context) {
+        upgrade(connection, host, attachment, null, context);
+    }
+
+    @Override
+    public void upgrade(
+            final ManagedAsyncClientConnection connection,
+            final HttpHost host,
+            final Object attachment,
+            final FutureCallback<ManagedAsyncClientConnection> callback,
+            final HttpContext context) {
```

Comment:
@reta For the sake of API consistency, could you please keep the callback as the last parameter? Otherwise looks good to me.

Reply:
👍 Sure thing, thank you!